pkg/autoupgrade/daemon.go

```diff
@@ -222,6 +224,17 @@ func (d *daemon) refreshImages(ctx context.Context, apps map[kclient.ObjectKey]v
 			}
 
 			if updated || strings.TrimPrefix(app.Status.AppImage.Digest, "sha256:") != strings.TrimPrefix(digest, "sha256:") {
+				if !updated && digest != "" {
```

Comment:
why do you need to check for a non empty digest here?

Reply:
We don't **need** to.
However, if the digest is empty, that means that we didn't find one above, thus the `checkImageAllowed` function should also fail or just be unnecessary, because it's working on the digest.
Low impact though.